include/aws/iotdevice/secure_tunneling.h

```diff
@@ -12,114 +12,275 @@
 
-AWS_IOTDEVICE_API
-int aws_secure_tunnel_connect(struct aws_secure_tunnel *secure_tunnel);
-
-AWS_IOTDEVICE_API
-int aws_secure_tunnel_close(struct aws_secure_tunnel *secure_tunnel);
-
-AWS_IOTDEVICE_API
-int aws_secure_tunnel_send_data(struct aws_secure_tunnel *secure_tunnel, const struct aws_byte_cursor *data);
-
-AWS_IOTDEVICE_API
-int aws_secure_tunnel_stream_start(struct aws_secure_tunnel *secure_tunnel);
-
-AWS_IOTDEVICE_API
-int aws_secure_tunnel_stream_reset(struct aws_secure_tunnel *secure_tunnel);
-
 /**
- * Raises exception and returns AWS_OP_ERR if options are missing required parameters.
+ * Asynchronous notify to the secure tunnel that you want it to attempt to connect.
+ * The secure tunnel will attempt to stay connected.
+ *
+ * @param secure_tunnel secure tunnel to start
+ * @return success/failure in the synchronous logic that kicks off the start process
  */
 AWS_IOTDEVICE_API
-int aws_secure_tunnel_options_validate(const struct aws_secure_tunnel_options *options);
+int aws_secure_tunnel_start(struct aws_secure_tunnel *secure_tunnel);
 
 /**
- * Create persistent storage for aws_secure_tunnel_options.
- * Makes a deep copy of (or acquires reference to) any data referenced by options,
+ * Asynchronous notify to the secure tunnel that you want it to transition to the stopped state. When the
+ * secure tunnel reaches the stopped state, all session state is erased.
+ *
+ * @param secure_tunnel secure tunnel to stop
+ * @return success/failure in the synchronous logic that kicks off the start process
  */
 AWS_IOTDEVICE_API
-struct aws_secure_tunnel_options_storage *aws_secure_tunnel_options_storage_new(
-    const struct aws_secure_tunnel_options *options);
+int aws_secure_tunnel_stop(struct aws_secure_tunnel *secure_tunnel);
 
 /**
- * Destroy options storage, and release any references held.
+ * Queues a message operation in a secure tunnel
+ *
+ * @param secure_tunnel secure tunnel to queue a message for
+ * @param message_options configuration options for the message operation
+ * @return success/failure in the synchronous logic that kicks off the message operation
  */
 AWS_IOTDEVICE_API
-void aws_secure_tunnel_options_storage_destroy(struct aws_secure_tunnel_options_storage *storage);
+int aws_secure_tunnel_send_message(
+    struct aws_secure_tunnel *secure_tunnel,
+    const struct aws_secure_tunnel_message_view *message_options);
 
-/**
- * Return pointer to options struct stored within.
- */
+//***********************************************************************************************************************
+/* THIS API SHOULD ONLY BE USED FROM SOURCE MODE */
+//***********************************************************************************************************************
 AWS_IOTDEVICE_API
-const struct aws_secure_tunnel_options *aws_secure_tunnel_options_storage_get(
-    const struct aws_secure_tunnel_options_storage *storage);
+int aws_secure_tunnel_stream_start(
```

Comment:
Stream Start is a specific operation that's processed differently than a message. It obtains the stream id differently and I'm assuming this will need to be expanded further for V3.

I thought about moving them to a private header but we still need to support this in a way that prevents breaking existing use in CPP because it was exposed. This one feels like a gray area in terms of how we want to handle it but for now, I'll leave it here since it needs to be accessible by the CPP client.